hazelcast/src/main/java/com/hazelcast/instance/GroupProperty.java

```diff
@@ -548,6 +548,18 @@
      */
     QUERY_OPTIMIZER_TYPE("hazelcast.query.optimizer.type", QueryOptimizerFactory.Type.RULES.toString()),
 
+
+    /**
+     * Cache de-serialized entries. Decreases number of de-serializations needed to evaluate a query when true.
+     * It allows to trade some memory for CPU cycles.
+     *
+     * Default: true
```

Comment:
I would probably go for speed by default.
Do we have other flags that have similar trade-offs (speed vs. memory-consumption)? 
It would be good to have consistent default values among all of them.